index.js

```diff
@@ -2613,6 +2613,42 @@
   }
   S.ifElse = def('ifElse', {}, [Pred(a), Fn(a, b), Fn(a, b), a, b], ifElse);
 
+  //# when :: (a -> Boolean) -> (a -> b) -> a -> b
```

Comment:
Great catch, Irakli! Returning `Either a b` seems nice, but if we do so the functions will not be sugar for `ifElse` with `I` in second or third position. The `Either a b` version may also be worth adding to the library, but we should wait to discover a compelling use case. :)